src/main/java/com/nutomic/syncthingandroid/syncthing/EventProcessor.java

```diff
@@ -0,0 +1,90 @@
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * As soon as the WebGui is available this event processor starts polling the event API.
+ * All events are converted to broadcast intents within the base URL com.nutomic.syncthingandroid.event.
+ * The type of the event is appended to the base URL to form the intent name. All parameters are
+ * passed along with the event as a parcelable.
+ * Which events are processed is defined within @see(RestApi.getEvents).
+ */
+public class EventProcessor implements SyncthingService.OnWebGuiAvailableListener {
+
+    private static final String TAG = "EventProcessor";
+
+    private static final String EVENT_BASE_URL = "com.nutomic.syncthingandroid.event";
+
+    /**
+     * Minimum interval in ms at which the events are polled from syncthing and processed.
+     * This intervall will not wake up the device to save battery power.
+     */
+    public static final int EVENT_UPDATE_INTERVAL = 10000;
+
+    private final ScheduledExecutorService mScheduledExecutor =  Executors.newScheduledThreadPool(1);
+    private ScheduledFuture<?> mEventPollerFuture;
+
+    private final Context mContext;
+    private final RestApi mApi;
+
+    public EventProcessor(Context context, RestApi api) {
+        mContext = context;
+        mApi = api;
+    }
+
+    @Override
+    public void onWebGuiAvailable() {
+        mEventPollerFuture = mScheduledExecutor.scheduleAtFixedRate(new Runnable() {
+
+            private long mLastEventId = 0;
+            private AsyncState currentState = null;
+
+            @Override
+            public void run() {
+                // Only read the events if the last call already returned.
+                if ((currentState == null) || (!currentState.isRunning()))
+                {
+                    Log.d(TAG, "Reading events starting with id " + mLastEventId);
+
+                    currentState = mApi.getEvents(mLastEventId, new RestApi.OnReceiveEventListener() {
+                        @Override
+                        public void onEvent(final long id, final String eventType, final Bundle eventData) {
+                            Intent intent = new Intent();
+                            intent.setAction(EVENT_BASE_URL + "." + eventType);
+                            intent.replaceExtras(eventData);
+
+                            mContext.sendBroadcast(intent);
+
+                            Log.d(TAG, "Event broadcast for " + eventType);
+                        }
+
+                        @Override
+                        public void onDone(long id) {
+                            if (mLastEventId < id) mLastEventId = id;
+                        }
+                    });
+                }
+            }
+        }, EVENT_UPDATE_INTERVAL, EVENT_UPDATE_INTERVAL, TimeUnit.MILLISECONDS);

```

Comment:
I was a little unsure if starting to process events asap would interfere with the other start-up calls. So I backed off a little.